File: Data Structures & Algorithms/two-integer-sum/submission-0.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        d = {}

        for idx, x in enumerate(nums):
            diff = target - x
            if diff in d:
                return [d[diff], idx]
            else:
                if not x in d:
                    d[x] = idx

        

File: Data Structures & Algorithms/two-integer-sum-ii/submission-0.cs
public class Solution {
    public int[] TwoSum(int[] numbers, int target) {
        var i = 0;
        var j = numbers.Length - 1;

        while (i < j) {
            var currSum = numbers[i] + numbers[j];
            if (currSum == target) {
                return [i + 1, j + 1] ;
            }

            else if (currSum < target) {
                i++;
            } else {
                j--;
            }
        }

        return [i + 1, j + 1];
    }
}
